File: docs/code-examples/tensor_one_dim.rs
//! Create and log a one dimensional tensor.

use ndarray::{Array, ShapeBuilder};
use rand::{thread_rng, Rng};
use rand_distr::StandardNormal;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) = rerun::RecordingStreamBuilder::new("rerun_example_tensors").memory()?;

    let mut data = Array::<f64, _>::default((100).f());
    data.map_inplace(|x| *x = thread_rng().sample(StandardNormal));

    rec.log(
        "tensor",
        &rerun::Tensor::try_from(data.as_standard_layout().view())?,
    )?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}

File: docs/code-examples/tensor_one_dim.py
"""Create and log a one dimensional tensor."""

import rerun as rr
from numpy.random import default_rng

rng = default_rng(12345)
tensor = rng.laplace(0.0, 1.0, 100)  # 1-dimensional tensor

rr.init("rerun_example_tensors", spawn=True)

# Log the tensor, assigning names to each dimension
rr.log_tensor("tensor", tensor)
